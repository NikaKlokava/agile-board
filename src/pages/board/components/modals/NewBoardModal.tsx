import { Button } from "../../../../shared/components/button";
import { FieldWrapper } from "../../../../shared/components/field_wrapper";
import { ModalWrapper } from "../../../../shared/components/modal_wrapper";
import { Formik } from "formik";
import { addBoard } from "../../../../redux/actionCreators/newBoardCreator";
import { Input } from "../../../../shared/components/input";
import { useState } from "react";
import { initialData } from "../../../../utils/utils";
import { useDispatch } from "react-redux";
import cl from "./modal_styles.module.css";

type Props = {
  onClose: () => void;
};

export const NewBoardModal = ({ onClose }: Props) => {
  const [columns, setColumns] = useState<number>(2);
  const dispatch = useDispatch();
  return (
    <ModalWrapper onWrapperClick={onClose}>
      <Formik
        initialValues={initialData}
        onSubmit={(values) => {
          dispatch(
            addBoard({
              name: values.name!,
              board_columns: values.board_columns,
              selected: false,
            })
          );
          onClose();
        }}
      >
        {(props) => (
          <>
            <h2 className={cl.modal_title} data-testid="new-board-modal">
              Add New Board
            </h2>
            <FieldWrapper fieldName={"Board Name"}>
              <input
                type={"text"}
                placeholder="e.g Take coffee break"
                spellCheck={false}
                className={cl.input_style}
                autoComplete="off"
                name="name"
                onChange={props.handleChange}
              />
            </FieldWrapper>
            <FieldWrapper fieldName={"Board Columns"} clName="style_container">
              {Array.from({ length: columns }, (_, index) => (
                <Input index={index} key={index} />
              ))}
            </FieldWrapper>
            <Button
              text={"Add New Column"}
              withIcon={true}
              testid={"add-new-column-btn"}
              onClick={() => setColumns((prev) => prev + 1)}
            />
            <Button
              text={"Create New Board"}
              withIcon={false}
              newClass="center"
              testid="create-new-board-btn"
              onClick={() => {
                props.handleSubmit();
              }}
            />
          </>
        )}
      </Formik>
    </ModalWrapper>
  );
};
