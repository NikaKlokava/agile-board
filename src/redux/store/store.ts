import { legacy_createStore } from "redux";
import { boardsReducer } from "../reducers/boardReducer";

const store = legacy_createStore(boardsReducer);

export default store;

// export type RootState = ReturnType<typeof store.getState>

// export interface RootState {
//     user: UserState;
//     loader: LoaderState;
//   }
// export type AppDispatch = typeof store.dispatch
