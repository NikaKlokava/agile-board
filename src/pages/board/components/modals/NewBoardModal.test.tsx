import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../../../../redux/store/store";
import { NewBoardModal } from "./NewBoardModal";

describe("Test the NewBoardModal component", () => {
  test("The NewBoardModal renders correctly", () => {
    const newBoardModalSnap = renderer
      .create(
        <Provider store={store}>
          <NewBoardModal onClose={() => console.log("close")} />
        </Provider>
      )
      .toJSON();
    expect(newBoardModalSnap).toMatchSnapshot();
  });
});
