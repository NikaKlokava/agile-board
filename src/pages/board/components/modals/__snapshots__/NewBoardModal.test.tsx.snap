// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the NewBoardModal component The NewBoardModal renders correctly 1`] = `
Array [
  <div
    className="modal_wrapper"
    onClick={[Function]}
  />,
  <div
    className="modal_container"
  >
     
    <h2
      className="modal_title"
      data-testid="new-board-modal"
    >
      Add New Board
    </h2>
    <div
      className="container"
    >
      <p
        className="title"
      >
        Task name
      </p>
      <input
        autoComplete="off"
        name="name"
        onChange={[Function]}
        placeholder="e.g Take coffee break"
        spellCheck={false}
        type="text"
      />
    </div>
    <div
      className="container style_container"
    >
      <p
        className="title"
      >
        Board Columns
      </p>
      <div
        className="input_container"
      >
        <input
          autoComplete="off"
          className="input_style"
          maxLength={14}
          name="columns[0].title"
          onChange={[Function]}
          placeholder="e.g Take coffee break"
          spellCheck={false}
          type="text"
        />
        <div
          className="delete_icon"
          onClick={[Function]}
        />
      </div>
      <div
        className="input_container"
      >
        <input
          autoComplete="off"
          className="input_style"
          maxLength={14}
          name="columns[1].title"
          onChange={[Function]}
          placeholder="e.g Take coffee break"
          spellCheck={false}
          type="text"
        />
        <div
          className="delete_icon"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="button"
      data-testid="add-new-column-btn"
      onClick={[Function]}
    >
      <div
        className="button_icon"
      />
      <div
        className="button_text"
      >
        Add New Column
      </div>
    </div>
    <div
      className="button center"
      data-testid="create-new-board-btn"
      onClick={[Function]}
    >
      <div
        className="button_text"
      >
        Create New Board
      </div>
    </div>
  </div>,
]
`;
